src/rust/engine/src/externs/fs.rs

```diff
@@ -269,22 +269,22 @@ impl PyAddPrefix {
   #[new]
   fn __new__(digest: PyDigest, prefix: PathBuf) -> Self {
     Self {
-      digest: digest.0.todo_as_digest(),
+      digest: digest.0,
       prefix,
     }
   }
 
   fn __hash__(&self) -> u64 {
     let mut s = DefaultHasher::new();
-    self.digest.hash.prefix_hash().hash(&mut s);
+    self.digest.as_digest().hash.prefix_hash().hash(&mut s);
```

Comment:
This naming is a bit confusing. Should it be `self.directory_digest`? Note that we can have the Rust strict use different names than what Python sees.